### PR TITLE
Enable Encrypted json export of vaults

### PR DESCRIPTION
**Pull request by MGibson1**
# Overview

Allow for exporting vaults in an encrypted state.

# File changes

* **ExportVaultPage.xaml/ExportVaultPage.xaml.cs/ExportVaultPageViewModel.cs**: Add binding to update export warning message to appropriate message depending on export format.
* **AppResources.resx**: Add encrypted export warning message (matches jslib's)
* **Models/Export/\***: Update constructors to allow export of encrypted Domain types.

# Image

image

### CODE REVIEW

File: src/App/Pages/Settings/ExportVaultPage.xaml.cs [hunk 1]
```diff
@@ -65,5 +65,13 @@ private async void ExportVault_Clicked(object sender, EventArgs e)
                 await _vm.ExportVaultAsync();
             }
         }
+
+        void _fileFormatPicker_SelectedIndexChanged(object sender, EventArgs e)
```
Comment (mportune-bw) [inline src/App/Pages/Settings/ExportVaultPage.xaml.cs:69]: Method naming convention is irregular here, maybe use something like `FileFormat_Changed`

**Comment by MGibson1** [comment]
Note, jslib export nulls our Cipher CollectionIds prior to export: https://github.com/bitwarden/jslib/blob/master/src/services/export.service.ts#L121

while mobile does not (nor does this implementation)

**Comment by mportune-bw** [comment]
> Note, jslib export nulls our Cipher CollectionIds prior to export: https://github.com/bitwarden/jslib/blob/master/src/services/export.service.ts#L121
> 
> while mobile does not (nor does this implementation)

Good catch, looks like an oversight on my part.  Go ahead and add that to mobile so the export results remain identical with other clients.

Looks legit otherwise 👍

File: src/App/Pages/Settings/ExportVaultPage.xaml.cs [hunk 2]
```diff
@@ -65,5 +65,13 @@ private async void ExportVault_Clicked(object sender, EventArgs e)
                 await _vm.ExportVaultAsync();
             }
         }
+
+        void FileFormat_Changed(object sender, EventArgs e)
+        {
+            if (_vm != null)
+            {
+                _vm.UpdateWarning();
+            }
```
Comment (kspearrin) [inline src/App/Pages/Settings/ExportVaultPage.xaml.cs:74]:
```suggestion
            _vm?.UpdateWarning();
```

**Review by mportune-bw — APPROVED** [review]

**Review by kspearrin — APPROVED** [review]

**Comment by xtqqczze** [comment]
Might this fix #798?

**Comment by MGibson1** [comment]
@xtqqczze It is not specifically related, no. This is a new file format to export a vault in that is encrypted in such a way that Bitwarden can import it again.

**Pull request merged**